Mobile Buy SDK/Mobile Buy SDK/Models/BUYAccountCredentials.m

```diff
@@ -26,94 +26,133 @@
 
 #import "BUYAccountCredentials.h"
 
-@class BUYAccountCredentialItem;
+NSString * const BUYAccountFirstNameKey            = @"first_name";
+NSString * const BUYAccountLastNameKey             = @"last_name";
+NSString * const BUYAccountEmailKey                = @"email";
+NSString * const BUYAccountPasswordKey             = @"password";
+NSString * const BUYAccountPasswordConfirmationKey = @"password_confirmation";
+
+#pragma mark - BUYAccountCredentials -
 @interface BUYAccountCredentials()
+
 @property (strong, nonatomic) NSMutableDictionary<NSString *, BUYAccountCredentialItem *> *items;
+
 @end
 
 @implementation BUYAccountCredentials
 
++ (BUYAccountCredentials *)credentials
+{
+	return [BUYAccountCredentials credentialsWithItems:nil];
+}
+
 + (BUYAccountCredentials *)credentialsWithItems:(NSArray<BUYAccountCredentialItem *> *)items
 {
-	BUYAccountCredentials *credentials = [BUYAccountCredentials new];
-	NSMutableDictionary *keyedItems = [NSMutableDictionary dictionary];
-	for (BUYAccountCredentialItem *item in items) {
-		keyedItems[item.key] = item;
+	return [[BUYAccountCredentials alloc] initWithItems:items];
+}
+
+- (instancetype)initWithItems:(NSArray<BUYAccountCredentialItem *> *)items
+{
+	self = [super init];
+	if (self) {
+		
+		_items = [NSMutableDictionary new];
+		if (items.count > 0) {
+			[self setCredentialItems:items];
+		}
 	}
-	credentials.items = keyedItems;
-	return credentials;
+	return self;
+}
+
+#pragma mark - Accessors -
+- (NSUInteger)count
+{
+	return self.items.count;
+}
+
+- (BOOL)isValid
+{
+	__block BOOL valid = YES;
+	[self.items enumerateKeysAndObjectsUsingBlock:^(NSString *key, BUYAccountCredentialItem *item, BOOL * _Nonnull stop) {
+		if (!item.isValid) {
+			valid = NO;
+			*stop = YES;
+		}
+	}];
+	return valid;
 }
 
-+ (BUYAccountCredentials *)credentialsWithItemKeys:(NSArray<NSString *> *)keys
+- (void)setCredentialItems:(NSArray<BUYAccountCredentialItem *> *)items
 {
-	NSMutableArray *items = [NSMutableArray array];
-	for (NSString *key in keys) {
-		BUYAccountCredentialItem *item = [BUYAccountCredentialItem itemWithKey:key value:@""];
-		[items addObject:item];
+	NSAssert(items, @"BUYAccountCredentialItem array cannot be nil.");
+	for (BUYAccountCredentialItem *item in items) {
+		[self setCredentialItem:item];
 	}
-	return [BUYAccountCredentials credentialsWithItems:items];
 }
 
+- (void)setCredentialItem:(BUYAccountCredentialItem *)item
```

Comment:
I completely agree with immutability here. The only reason I kept it is to maintain a similar API to what it was before. Otherwise, I'm happy to make this an immutable class.